packages/core/src/components/button/button.tsx

```diff
@@ -70,6 +76,8 @@ export class TdsButton {
       >
         <button
           type={this.type}
+          name={this.name}
+          value={this.value}
```

Comment:
Since the value attribute is used during form submission, it's undesirable to include it if its value might be undefined. A more clean approach would be to conditional render it

```
<button
  type={this.type}
  {...(this.value ? { value: this.value } : {})}
  ...
  ```